enterprise/reporting/reporting.go

```diff
@@ -218,13 +221,16 @@ func (r *HandleT) getReports(currentMs int64, clientName string) (reports []*typ
 	}
 
 	queryStart := time.Now()
-	err = dbHandle.QueryRow(sqlStatement).Scan(&queryMin)
-	if err != nil && err != sql.ErrNoRows {
+	ctx, cancel := context.WithTimeout(context.Background(), r.dbQueryTimeout)
+	defer cancel()
+	err = dbHandle.QueryRowContext(ctx, sqlStatement).Scan(&queryMin)
+
+	if err != nil && err != sql.ErrNoRows && err != context.DeadlineExceeded {
```

Comment:
In case `err != sql.ErrNoRows` should we return `err = nil` or use a custom error:

```suggestion
    if err != sql.ErrNoRows {
          return nil, 0, ErrNoReports
    }
	if err != nil && err != context.DeadlineExceeded {
```

It is not a good idea to expose `sql.ErrNoRows` to the rest of the system, since it is implementation/SQL specific.

Reply:
returning custom error is query timeouts and returning nil if there are 0 reports